src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -161,10 +161,26 @@ void remove_redundant_reorders::run(program& p) {
             continue;
         }
 
+        auto no_output_padding = [&](reorder_node& node) -> bool {
+            auto output_padding = node.get_output_layout().data_padding;
+            auto u_pad = output_padding.upper_size().sizes();
+            auto l_pad = output_padding.lower_size().sizes();
+            for (auto& p : u_pad) {
+                if (p != 0)
+                    return false;
+            }
+            for (auto& p : l_pad) {
+                if (p != 0)
+                    return false;
+            }
+            return true;
+        };
+
         bool remove_current = r_node.is_simple_reorder() &&
                               r_dep_node.get_users().size() == 1 &&
                               !r_dep_node.is_output() &&
-                              !r_node.get_primitive()->has_surface_input();
+                              !r_node.get_primitive()->has_surface_input() &&
+                              no_output_padding(r_node);
```

Comment:
We can use `bool cldnn::padding::operator bool()` instead of `no_output_padding`, i.e:
```suggestion
                              !r_node.get_output_layout().data_padding;
```

Reply:
BTW, maybe we need to execute reorder only in case if `input_padding != output_paddding`?

Reply:
input_padding != output_paddding is better idea! Thanks.